pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go

```diff
@@ -1168,6 +1168,7 @@ func (rc *rangeController) HandleRaftEventRaftMuLocked(ctx context.Context, e Ra
 			}
 		}
 		shouldWaitChange = rs.handleReadyEntriesRaftMuLocked(ctx, rs.scratchEvent, rd) || shouldWaitChange
+		rs.scratchEvent = raftEventForReplica{}
```

Comment:
Can probably also drop line 1072, since the invariant is now that `scratchEvent` is always empty except within this method.